### PR TITLE
Implement cmath python module

### PR DESCRIPTION
**Pull request by Smit-create**
#200

### CODE REVIEW

File: src/runtime/impure/lfortran_intrinsics.c [hunk 1]
```diff
@@ -225,6 +225,16 @@ LFORTRAN_API double _lfortran_dlog10(double x)
     return log10(x);
 }
 
+LFORTRAN_API float_complex_t _lfortran_clog10(float_complex_t x)
+{
+    return log10(x);
```
Comment (certik) [inline src/runtime/impure/lfortran_intrinsics.c:230]: This line fails to compile on Windows with the error:
```
D:\a\lpython\lpython\lpython-0.1.0-788-gcb5b5f606\src\runtime\impure\lfortran_intrinsics.c(230): error C2440: 'function': cannot convert from 'float_complex_t' to 'double'
```

Comment (certik) [reply src/runtime/impure/lfortran_intrinsics.c:230]: Typically those complex functions are prefixed with `c`.

Comment (certik) [reply src/runtime/impure/lfortran_intrinsics.c:230]: Also for single precision, they use the `f` suffix.

However, it seems that `log10` does not have a complex version? The `clog10` and `clog10f` don't seem to compile for me.

File: integration_tests/test_cmath.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from ltypes import c64
+
+def test_power_logarithmic():
+    x: c64
+    y: c64
+    x = complex(3, 3)
+    y = exp(x)
+    y = log(x)
+    y = sqrt(x)
+    y = log10(x)
```
Comment (certik) [inline integration_tests/test_cmath.py:12]: Let's also test the results in `y` to make sure we are getting the right answer and not 0. I tested locally and it seems to work. I think we need to implement this in LPython first:
```
>>> y = 3+3j
>>> y
(3+3j)
>>> y.real
3.0
>>> y.imag
3.0
```

And absolute value for complex numbers:
```
>>> abs(y - (3+3j))
0.0
>>> abs(y - (3+3.001j))
0.0009999999999998899
```

So let's do it in subsequent PRs.

File: src/runtime/impure/lfortran_intrinsics.c (hunk 1 above)
Comment (certik) [reply src/runtime/impure/lfortran_intrinsics.c:230]: So if this is needed in Python, you can compute it from Python using the complex log of base `e`.

Comment (Smit-create) [reply src/runtime/impure/lfortran_intrinsics.c:230]: > you can compute it from Python using the complex log of base `e`.

Yeah, that should work too. Thanks!

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
This looks good, thanks! Let's get the real/imag parts of complex numbers working, so that we can add tests here that test the results.

**Comment by Smit-create** [comment]
Yep, sounds good!

**Pull request merged**